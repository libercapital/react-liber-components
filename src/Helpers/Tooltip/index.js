export { default } from './Tooltip';
export { default as TooltipBox } from './TooltipBox';
