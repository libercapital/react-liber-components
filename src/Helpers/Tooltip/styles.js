import styled, { css } from 'styled-components';
import { handleThemeFromObject as getTheme } from '../../Themes';
import { FaQuestionCircle } from '../../Icons';

const arrowSize = 18;

const CONTENT_MOUNT = {
  start: { percentage: 0, arrowSubtraction: 0, arrowOffset: arrowSize },
  middle: { percentage: 50, arrowSubtraction: 18, arrowOffset: 0 },
  end: {
    percentage: 100,
    arrowSubtraction: 36,
    arrowOffset: -1 * arrowSize,
  },
  unset: { percentage: 50, arrowSubtraction: 18, arrowOffset: 0 },
};

export const Container = styled.div`
  position: relative;
  width: 20px;
  height: 20px;
  font-family: inherit;
  font-size: 12px;
  font-weight: 500;
  fill: ${props => getTheme(props.theme, 'tooltip.fill', '#a4a4a4')};
  color: ${props => getTheme(props.theme, 'tooltip.color', '#717171')};
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    fill: ${props => getTheme(props.theme, 'tooltip.hover', '#009DFF')};
  }

  @media (max-width: 992px) {
    width: 28px;
    height: 28px;
  }
`;

const positionFloat = props => {
  const { position, contentPosition } = props;
  const { arrowOffset, percentage } = CONTENT_MOUNT[contentPosition];
  switch (position) {
    /** floater tooltip styles */
    case 'mouse-right':
      return css`
        right: auto;
        transform: translate(${props.mousePadding}px, -50%);
      `;
    case 'mouse-left':
      return css`
        left: auto;
        transform: translate(calc(-100% - ${props.mousePadding}px), -50%);
      `;
    case 'mouse-top':
      return css`
        top: auto;
        transform: translate(-50%, calc(-100% - ${props.mousePadding}px));
      `;
    case 'mouse-bottom':
      return css`
        right: auto;
        transform: translate(-50%, ${props.mousePadding}px);
      `;
    case 'content-right':
      return css`
        right: auto;
        transform: translate(${props.mousePadding}px, calc(-${percentage}% - ${arrowOffset}px));
      `;
    case 'content-left':
      return css`
        left: auto;
        transform: translate(
          calc(-100% - ${props.mousePadding}px),
          calc(-${percentage}% - ${arrowOffset}px)
        );
      `;
    case 'content-top':
      return css`
        top: auto;
        transform: translate(
          calc(-${percentage}% - ${arrowOffset}px),
          calc(-100% - ${props.mousePadding}px)
        );
      `;
    case 'content-bottom':
      return css`
        bottom: auto;
        transform: translate(calc(-${percentage}% - ${arrowOffset}px), ${props.mousePadding}px);
      `;
    /* fixed tooltip styles */
    case 'bottom-right':
      return css`
        left: 0px;
        right: auto;
        top: ${props.visible ? '0px' : '10px'};
        transform: translate(30px, 30px);
      `;
    case 'bottom-center':
      return css`
        top: ${props.visible ? '0px' : '10px'};
        transform: translate(0px, 30px);
      `;
    case 'bottom-left':
      return css`
        left: auto;
        right: 0px;
        top: ${props.visible ? '0px' : '10px'};
        transform: translate(-30px, 30px);
      `;
    case 'center-right':
      return css`
        bottom: 50%;
        right: auto;
        left: ${props.visible ? '0px' : '10px'};
        transform: translate(30px, 30px);
      `;
    case 'center-left':
      return css`
        bottom: 50%;
        left: auto;
        right: ${props.visible ? '0px' : '-10px'};
        transform: translate(-30px, 30px);
      `;
    case 'top-right':
      return css`
        left: 0px;
        right: auto;
        bottom: ${props.visible ? '0px' : '-10px'};
        transform: translate(30px, -30px);
      `;
    case 'top-center':
      return css`
        bottom: ${props.visible ? '0px' : '-10px'};
        transform: translate(0px, -30px);
      `;
    case 'top-left':
    default:
      return css`
        left: auto;
        right: 0px;
        bottom: ${props.visible ? '0px' : '-10px'};
        transform: translate(-30px, -30px);
      `;
  }
};

const positionArrow = props => {
  const { mount, contentPosition } = props;
  const { percentage, arrowSubtraction } = CONTENT_MOUNT[contentPosition];

  switch (mount) {
    case 'left':
      return css`
        width: 5px !important;
        height: 36px !important;
        top: calc((${percentage}%) - ${arrowSubtraction}px);
        left: 100%;
      `;
    case 'top':
      return css`
        width: 36px !important;
        height: 5px !important;
        left: calc((${percentage}%) - ${arrowSubtraction}px);
        top: 100%;
      `;
    case 'bottom':
      return css`
        width: 36px !important;
        height: 5px !important;
        left: calc((${percentage}%) - ${arrowSubtraction}px);
        bottom: 100%;
      `;
    case 'right':
    default:
      return css`
        width: 5px !important;
        height: 36px !important;
        top: calc((${percentage}%) - ${arrowSubtraction}px);
        right: 100%;
      `;
  }
};
const positionArrowAfter = props => {
  switch (props.mount) {
    case 'left':
      return css`
        top: calc(50% - 7.5px);
        right: calc(100% - 7.5px);
      `;
    case 'top':
      return css`
        left: calc(50% - 7.5px);
        bottom: calc(100% - 7.5px);
      `;
    case 'bottom':
      return css`
        left: calc(50% - 7.5px);
        top: calc(100% - 7.5px);
      `;
    case 'right':
    default:
      return css`
        top: calc(50% - 7.5px);
        left: calc(100% - 7.5px);
      `;
  }
};

export const Arrow = styled.div`
  position: absolute;
  ${props => positionArrow(props)}
  overflow: hidden;
  pointer-events: none;
  padding: 8px;
  box-sizing: border-box;

  &::after {
    content: '';
    position: absolute;
    pointer-events: none;
    width: 15px;
    height: 15px;
    background: #fff;
    transform: rotate(45deg);
    ${props => positionArrowAfter(props)}
    box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.2);
  }
`;

const mousePositions = [
  'mouse-right',
  'mouse-left',
  'mouse-top',
  'mouse-bottom',
  'content-right',
  'content-left',
  'content-top',
  'content-bottom',
];

export const Float = styled.div.attrs(({ position, x, y }) => ({
  style: {
    left: mousePositions.includes(position) ? x : undefined,
    top: mousePositions.includes(position) ? y : undefined,
  },
}))`
  position: absolute;
  border-radius: 4px;
  background-color: ${props => getTheme(props.theme, 'colors.v2.white', '#fff')};
  box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.2);
  z-index: 9999;
  display: block;
  pointer-events: ${props => (props.visible && !props.hoverable ? 'auto' : 'none')};
  opacity: ${props => (props.visible ? '1' : '0')};
  transition: ${props => (mousePositions.includes(props.position) ? null : 'all 0.1s ease')};
  color: ${props => getTheme(props.theme, 'colors.v2.grey4', '#6287a7')};
  font-size: ${props => getTheme(props.theme, 'fontSizes.mobileLarge', '14px')};
  font-weight: ${props => getTheme(props.theme, 'fontWeights.regular', '400')};
  line-height: 20px;

  ${props => positionFloat(props)}

  & * {
    pointer-events: none;
  }

  & > h6 {
    font-weight: bold;
    margin: 0px;
    opacity: 0.75;
    font-family: inherit;
    font-size: 14px;
    text-align: left;
    color: ${props => getTheme(props.theme, 'tooltip.floatPrimary', '#009DFF')};
    background-color: ${props => getTheme(props.theme, 'tooltip.floatBgSecondary', '#fafafa')};
    padding: 6px 0px;
    text-align: center;
    user-select: none;
  }
  & > div {
    display: flex;
    width: auto;
    justify-content: flex-start;
    align-items: center;
    padding: 12px;
    overflow: hidden;
  }
`;

export const Icon = styled(FaQuestionCircle)`
  width: inherit;
  height: inherit;
  cursor: pointer;
`;
