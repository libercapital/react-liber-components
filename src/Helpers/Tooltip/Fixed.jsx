import React from 'react';
import PropTypes from 'prop-types';
import { ClickOutside } from '../../Util';
import { Container, Icon, Float } from './styles';

class Fixed extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: false,
    };
  }

  handleClick = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleBlur = () => {
    this.setState({ visible: false });
  };

  handleFocus = () => {
    this.setState({ visible: true });
  };

  render = () => {
    const {
      children,
      hoverable,
      label,
      position,
      contentPosition,
      ...props
    } = this.props;
    const { visible } = this.state;
    return (
      <ClickOutside onOutsideClick={this.handleBlur}>
        <Container
          onClick={hoverable ? null : this.handleClick}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
          onMouseOver={hoverable ? this.handleFocus : null}
          onMouseLeave={hoverable ? this.handleBlur : null}
          {...props}
        >
          <Icon />
          <Float
            visible={visible}
            position={position}
            hoverable={hoverable}
            contentPosition={contentPosition}
          >
            <h6>{label}</h6>
            <div>{children}</div>
          </Float>
        </Container>
      </ClickOutside>
    );
  };
}

Fixed.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
  hoverable: PropTypes.bool,
  label: PropTypes.string,
  position: PropTypes.string,
  contentPosition: PropTypes.oneOf(['start', 'middle', 'end', 'unset']),
};

Fixed.defaultProps = {
  hoverable: false,
  label: '',
  position: 'top-center',
  contentPosition: 'unset',
};

export default Fixed;
