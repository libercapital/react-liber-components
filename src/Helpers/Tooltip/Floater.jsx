import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { Float, Arrow } from './styles';

class Floater extends React.Component {
  handleBlur = () => {
    this.setState({ visible: false });
  };

  handleFocus = () => {
    this.setState({ visible: !this.state.visible });
  };

  isContentPositionUnset = () => {
    const { contentPosition } = this.props;
    return contentPosition === 'unset';
  };

  render = () => {
    const {
      children,
      label,
      show,
      position,
      mousePadding,
      mount,
      contentPosition,
    } = this.props;

    return ReactDOM.createPortal(
      <Float
        visible={show}
        position={
          this.isContentPositionUnset() ? `mouse-${mount}` : `content-${mount}`
        }
        x={`${position.x ? position.x + window.scrollX : 0}px`}
        y={`${position.y ? position.y + window.scrollY : 0}px`}
        mousePadding={mousePadding > 13 ? mousePadding : 13}
        hoverable={show}
        contentPosition={contentPosition}
      >
        <Arrow mount={mount} contentPosition={contentPosition} />
        {label.length > 0 ? <h6>{label}</h6> : null}
        <div>{children}</div>
      </Float>,
      document.getElementById('tooltip'),
    );
  };
}

Floater.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.string,
  ]).isRequired,
  label: PropTypes.string,
  show: PropTypes.bool,
  position: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
  }),
  mousePadding: PropTypes.number,
  mount: PropTypes.oneOf(['top', 'right', 'left', 'bottom']),
  contentPosition: PropTypes.oneOf(['start', 'middle', 'end', 'unset']),
};

Floater.defaultProps = {
  label: '',
  show: true,
  position: {},
  mousePadding: 40,
  mount: 'right',
  contentPosition: 'unset',
};

export default Floater;
