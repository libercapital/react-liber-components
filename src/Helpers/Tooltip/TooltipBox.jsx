import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Tooltip from './Tooltip';

class TooltipBox extends Component {
  state = {
    showTooltip: false,
    position: {
      x: null,
      y: null,
    },
  };

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }

  setPosition = (x, y, action, event) => {
    const { delay, delayTime } = this.props;
    this.timeout = setTimeout(
      () => {
        this.setState({
          showTooltip: true,
          position: { x, y },
        });
      },
      delay ? delayTime : 0,
    );
    if (action) {
      action(event);
    }
  };

  handleMouseMove = (move, event) => {
    this.setPosition(event.clientX, event.clientY, move, event);
  };

  handleMouseEnter = (enter, event) => {
    const { mount } = this.props;
    const targetNode = this.child;
    const computedStyle = window.getComputedStyle(targetNode);
    const width = parseInt(computedStyle.getPropertyValue('width'), 10) || 0;
    const height = parseInt(computedStyle.getPropertyValue('height'), 10) || 0;
    const viewportOffset = targetNode.getBoundingClientRect();

    const { top } = viewportOffset;
    const { left } = viewportOffset;
    let x;
    let y;
    switch (mount) {
      case 'left':
        x = left;
        y = top + height / 2;
        break;
      case 'right':
        x = left + width;
        y = top + height / 2;
        break;
      case 'top':
        x = left + width / 2;
        y = top;
        break;
      case 'bottom':
        x = left + width / 2;
        y = top + height;
        break;
      default:
        x = left + width / 2;
        y = top + height / 2;
        break;
    }
    this.setPosition(x, y, enter, event);
  };

  handleMouseLeave = (leave, event) => {
    clearTimeout(this.timeout);
    this.setState({ showTooltip: false, position: {} }, () => {
      if (leave) leave(event);
    });
  };

  render() {
    const { showTooltip, position } = this.state;
    const {
      show,
      content,
      label,
      mount,
      fixed,
      mousePadding,
      contentPosition,
    } = this.props;
    const children = React.Children.only(this.props.children);
    const props = {
      ref: (el) => {
        this.child = el;
      },
      onMouseMove: !fixed
        ? (event) => {
            this.handleMouseMove(children.props.onMouseMove, event);
          }
        : null,
      onMouseEnter: fixed
        ? (event) => {
            this.handleMouseEnter(children.props.onMouseEnter, event);
          }
        : null,
      onMouseLeave: (event) => {
        this.handleMouseLeave(children.props.onMouseLeave, event);
      },
    };
    const clone = React.cloneElement(children, props);
    return (
      <Fragment>
        {showTooltip && show ? (
          <Tooltip
            type="floater"
            show={showTooltip && show}
            position={position}
            label={label}
            mount={mount}
            contentPosition={contentPosition}
            mousePadding={mousePadding}
          >
            {content}
          </Tooltip>
        ) : null}
        {clone}
      </Fragment>
    );
  }
}
TooltipBox.propTypes = {
  children: PropTypes.oneOfType([PropTypes.any]).isRequired,
  content: PropTypes.oneOfType([PropTypes.any]).isRequired,
  show: PropTypes.bool,
  mount: PropTypes.oneOf(['top', 'right', 'left', 'bottom']),
  fixed: PropTypes.bool,
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      text: PropTypes.string,
      color: PropTypes.string,
    }),
  ]),
  delay: PropTypes.bool,
  delayTime: PropTypes.number,
  mousePadding: PropTypes.number,
  contentPosition: PropTypes.oneOf(['start', 'middle', 'end', 'unset']),
};
TooltipBox.defaultProps = {
  show: true,
  mount: 'right',
  contentPosition: 'unset',
  label: '',
  fixed: false,
  delay: false,
  delayTime: 1500,
  mousePadding: null,
};

export default TooltipBox;
