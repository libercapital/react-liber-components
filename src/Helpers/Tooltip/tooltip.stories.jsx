import React from 'react';
import styled from 'styled-components';
import Tooltip from './Tooltip';

const TooltipContent = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 100px;
  box-sizing: border-box;
`;

const Content = styled.div`
  width: 128px;
  font-size: 16px;
`;

export default {
  title: 'Components 1.0|Helpers/Tooltip',
  parameters: {
    jest: ['Tooltip'],
  },
};

export const defaultStory = () => (
  <TooltipContent>
    <Tooltip />
  </TooltipContent>
);

defaultStory.story = {
  name: 'default',
};

export const withLabel = () => (
  <TooltipContent>
    <Tooltip label="Prazo de Vencimento">
      <Content>10 dias Ãºteis</Content>
    </Tooltip>
  </TooltipContent>
);
