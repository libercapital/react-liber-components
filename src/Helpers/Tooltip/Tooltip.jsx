import React from 'react';
import Fixed from './Fixed';
import Floater from './Floater';

const Tooltip = ({ type, ...props }) => {
  switch (type) {
    case 'floater':
      return <Floater {...props} />;
    case 'fixed':
    default:
      return <Fixed {...props} />;
  }
};

export default Tooltip;
