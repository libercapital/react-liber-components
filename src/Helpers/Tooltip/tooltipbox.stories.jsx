import React from 'react';
import styled from 'styled-components';
import TooltipBox from './TooltipBox';

const TooltipContent = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 100px;
  box-sizing: border-box;
`;

const FixedTootlipContent = styled(TooltipContent)`
  width: 50%;
  height: 100px;
  margin: 100px;
  margin-left: 200px;
  padding: 0px;
  border: 1px solid blue;
`;

const tooltipContentComponent = <TooltipContent>Conteudo</TooltipContent>;

export default {
  title: 'Components 1.0|Helpers/Tooltip Box',

  parameters: {
    jest: ['TooltipBox'],
  },
};

export const defaultStory = () => (
  <TooltipBox content={tooltipContentComponent}>
    <TooltipContent>Passe o mouse aqui!</TooltipContent>
  </TooltipBox>
);

defaultStory.story = {
  name: 'default',
};

export const defaultContentPositionStart = () => (
  <TooltipBox contentPosition="start" content={tooltipContentComponent}>
    <TooltipContent>Passe o mouse aqui!</TooltipContent>
  </TooltipBox>
);

defaultContentPositionStart.story = {
  name: 'default contentPosition = start',
};

export const defaultWithContentPositionMiddle = () => (
  <TooltipBox contentPosition="middle" content={tooltipContentComponent}>
    <TooltipContent>Passe o mouse aqui!</TooltipContent>
  </TooltipBox>
);

defaultWithContentPositionMiddle.story = {
  name: 'default with contentPosition = middle',
};

export const defaultContentPositionEnd = () => (
  <TooltipBox contentPosition="end" content={tooltipContentComponent}>
    <TooltipContent>Passe o mouse aqui!</TooltipContent>
  </TooltipBox>
);

defaultContentPositionEnd.story = {
  name: 'default contentPosition = end',
};

export const fixed = () => (
  <TooltipBox fixed content={tooltipContentComponent}>
    <FixedTootlipContent>Passe o mouse aqui!</FixedTootlipContent>
  </TooltipBox>
);

export const fixedWithContentPositionStart = () => (
  <TooltipBox fixed contentPosition="start" content={tooltipContentComponent}>
    <FixedTootlipContent>Passe o mouse aqui!</FixedTootlipContent>
  </TooltipBox>
);

fixedWithContentPositionStart.story = {
  name: 'fixed with contentPosition = start',
};

export const fixedWithContentPositionMiddle = () => (
  <TooltipBox fixed contentPosition="middle" content={tooltipContentComponent}>
    <FixedTootlipContent>Passe o mouse aqui!</FixedTootlipContent>
  </TooltipBox>
);

fixedWithContentPositionMiddle.story = {
  name: 'fixed with contentPosition = middle',
};

export const fixedWithContentPositionEnd = () => (
  <TooltipBox fixed contentPosition="end" content={tooltipContentComponent}>
    <FixedTootlipContent>Passe o mouse aqui!</FixedTootlipContent>
  </TooltipBox>
);

fixedWithContentPositionEnd.story = {
  name: 'fixed with contentPosition = end',
};

export const fixedWithContentPositionStartOnBottom = () => (
  <TooltipBox fixed contentPosition="start" mount="bottom" content={tooltipContentComponent}>
    <FixedTootlipContent>Passe o mouse aqui!</FixedTootlipContent>
  </TooltipBox>
);

fixedWithContentPositionStartOnBottom.story = {
  name: 'fixed with contentPosition = start on bottom',
};
