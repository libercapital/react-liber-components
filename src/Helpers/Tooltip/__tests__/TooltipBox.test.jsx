import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TooltipBox from '../TooltipBox';
import Tooltip from '../Tooltip';

describe('Test for Floater Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'tooltip');
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);

  beforeEach(() => {
    props = {
      show: true,
      label: 'label',
      content: <div>content</div>,
      children: <div id="child">mock</div>,
    };
  });

  it('should render correctly with children', () => {
    const wrapper = shallow(<TooltipBox {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render Tooltip on mouseMove', () => {
    const wrapper = shallow(<TooltipBox {...props} />);
    const child = wrapper.find('#child');
    child.simulate('mouseMove', { clientX: 0, clientY: 0 });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render Tooltip on mouseEnter', () => {
    const wrapper = shallow(<TooltipBox {...props} />);
    const child = wrapper.find('#child');
    child.simulate('mouseEnter');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should remove Tooltip on mouseLeave', () => {
    const wrapper = shallow(<TooltipBox {...props} />);
    wrapper.setState({ showTooltip: true, position: { x: 0, y: 0 } });
    const tooltip = wrapper.find(Tooltip);
    expect(tooltip.length).toBe(1);
    const child = wrapper.find('#child');
    child.simulate('mouseLeave');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should not show tooltip with show=false', () => {
    props.show = false;
    const wrapper = shallow(<TooltipBox {...props} />);
    const child = wrapper.find('#child');
    child.simulate('mouseMove', { clientX: 0, clientY: 0 });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call clearTimeout on component unmount', () => {
    jest.useFakeTimers();
    const wrapper = shallow(<TooltipBox {...props} />);
    wrapper.instance().componentWillUnmount();
    expect(clearTimeout).toBeCalled();
  });

  it('should render correctly being fixed and with contentPostion ', () => {
    const wrapper = shallow(<TooltipBox fixed contentPosition="start" {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
