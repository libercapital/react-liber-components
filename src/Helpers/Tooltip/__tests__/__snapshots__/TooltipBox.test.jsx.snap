// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Floater Component should not show tooltip with show=false 1`] = `
<Fragment>
  <div
    id="child"
    onMouseEnter={null}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    mock
  </div>
</Fragment>
`;

exports[`Test for Floater Component should remove Tooltip on mouseLeave 1`] = `
<Fragment>
  <div
    id="child"
    onMouseEnter={null}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    mock
  </div>
</Fragment>
`;

exports[`Test for Floater Component should render Tooltip on mouseEnter 1`] = `
<Fragment>
  <div
    id="child"
    onMouseEnter={null}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    mock
  </div>
</Fragment>
`;

exports[`Test for Floater Component should render Tooltip on mouseMove 1`] = `
<Fragment>
  <div
    id="child"
    onMouseEnter={null}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    mock
  </div>
</Fragment>
`;

exports[`Test for Floater Component should render correctly being fixed and with contentPostion  1`] = `
<Fragment>
  <div
    id="child"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={null}
  >
    mock
  </div>
</Fragment>
`;

exports[`Test for Floater Component should render correctly with children 1`] = `
<Fragment>
  <div
    id="child"
    onMouseEnter={null}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    mock
  </div>
</Fragment>
`;
